src/libraries/Common/tests/SourceGenerators/RoslynTestUtils.cs

```diff
@@ -266,27 +266,51 @@ public static TextSpan MakeSpan(string text, int spanNum)
                 for (int i = 0; i < count; i++)
                 {
                     SourceText s = await proj.FindDocument(l[i]).GetTextAsync().ConfigureAwait(false);
-                    results.Add(s.ToString().Replace("\r\n", "\n", StringComparison.Ordinal));
+                    results.Add(Replace(s.ToString(), "\r\n", "\n"));
                 }
             }
             else
             {
                 for (int i = 0; i < count; i++)
                 {
                     SourceText s = await proj.FindDocument($"src-{i}.cs").GetTextAsync().ConfigureAwait(false);
-                    results.Add(s.ToString().Replace("\r\n", "\n", StringComparison.Ordinal));
+                    results.Add(Replace(s.ToString(), "\r\n", "\n"));
                 }
             }
 
             if (extraFile != null)
             {
                 SourceText s = await proj.FindDocument(extraFile).GetTextAsync().ConfigureAwait(false);
-                results.Add(s.ToString().Replace("\r\n", "\n", StringComparison.Ordinal));
+                results.Add(Replace(s.ToString(), "\r\n", "\n"));
             }
 
             return results;
         }
 
+        public static bool CompareLines(string[] expectedLines, SourceText sourceText, out string message)
+        {
+            if (expectedLines.Length != sourceText.Lines.Count)
+            {
+                message = string.Format("Line numbers do not match. Expected: {0} lines, but generated {1}",
+                    expectedLines.Length, sourceText.Lines.Count);
+                return false;
+            }
+            int index = 0;
+            foreach (TextLine textLine in sourceText.Lines)
+            {
+                string expectedLine = expectedLines[index];
+                if (!expectedLine.Equals(textLine.ToString(), StringComparison.Ordinal))
```

Comment:
>!expectedLine.Equals(textLine.ToString(), StringComparison.Ordinal)) [](http://example.com/codeflow?start=20&length=68)

nit: this can be as simple as 

```C#
if (expectedLine != textLine.ToString())
```

Reply:
I'm just porting this helper method to a shared location (with the logging generator). I'd prefer to keep the changes here minimal if that's okay.